src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -223,15 +223,24 @@ public override void DidReceiveResponse (NSUrlSession session, NSUrlSessionDataT
 				var inflight = GetInflightData (dataTask);
 
 				try {
+					// ensure that we did not cancel the request, if we did, do cancel the task
+					if (inflight.CancellationToken.IsCancellationRequested) {
+                            dataTask.Cancel();
+                    }
+
```

Comment:
^ incorrect alignment (spaces no tabs) and missing space before `()`. Please check your editor settings